utils/parser.py

```diff
@@ -320,55 +319,54 @@ def create_file_extractor(parse_type="pdf"):
     file_extractor = {".txt": parser}
     return file_extractor
 
-def has_markdown_tables(content):
-    """Check if content contains markdown tables"""
-    table_patterns = [
-        r'\|.*\|.*\|',          # Table row with cells
-        r'\|[\s-]*\|[\s-]*\|'   # Table header separator
-    ]
-    return all(re.search(pattern, content, re.MULTILINE) for pattern in table_patterns)
 
 def parse_txt_to_md(file_path, file_extension, title_tag=""):
     """
-    Parses a .txt file to a Markdown (.md) file using LlamaParse.
+    Parses a .txt file to a Markdown (.md) file.
+    It first checks if the content is already well-formed.
+    If not, it uses LlamaParse.
     """
-    # get the file extension
-
     with open(file_path, encoding="utf-8") as f:
-        content = f.read()
-
-        
-    if not has_markdown_tables(content):
-        documents = SimpleDirectoryReader(
-            input_files=[file_path], file_extractor=create_file_extractor(file_extension)
-        ).load_data()
-    else:
-        # save the content to a list of documents
-        documents = SimpleDirectoryReader(input_files=[file_path]).load_data()
+        content = f.read().strip()
 
-    # size = sum([len(doc.text) for doc in documents])
-    # validate if the content is empty
-    is_empty = all(is_empty_content(doc.text) for doc in documents)
-
-    # base_filename = os.path.basename(file_path)
     out_name = file_path.replace(".txt", ".md")
-
     title_tag = clean_title(title_tag)
 
-    # os.makedirs(os.path.dirname(out_name), exist_ok=True)
-
+    # Heuristic to check if content is already good markdown
```

Comment:
Keep the old if not has_markdown_tables(content) - call LlamaParse
and *add* if the content coming back from llamaparse is empty, then 
final_content = original content (txt file)
If the original content was also empty, then return True (fail)